src/scheduler/gc_work.rs

```diff
@@ -663,8 +663,12 @@ pub trait ProcessEdgesWork:
     /// trace the object and store back the new object reference if necessary.
     fn process_edge(&mut self, slot: EdgeOf<Self>) {
         let object = slot.load();
+        if object.is_null() {
+            return;
+        }
         let new_object = self.trace_object(object);
-        if Self::OVERWRITE_REFERENCE {
+        debug_assert!(!new_object.is_null());
+        if Self::OVERWRITE_REFERENCE && new_object != object {
```

Comment:
We don't need `OVERWRITE_REFERENCE` anymore. It's always `true` anyway.

Reply:
`SanityGCProcessEdges` sets that constant to `false`.

Reply:
There is no harm to check it, and we should check it unless we remove `OVERWRITE_REFERENCE`. Otherwise if there is a new implementation of `ProcessEdgesWork` that sets `OVERWRITE_REFERNECE` to `false`, we have a bug here.

Reply:
`OVERWRITE_REFERENCE` was a hack to implement sanity GC. Sanity GC does not move objects anyway and the check can be obviated now. I'm guessing the compiler is smart enough to figure out that the constant is either always true in the case of real GC work packets and always false for sanity, so there is no performance overhead of the check. So at the end of the day, it's fine to keep it, but I just think it's a bit redundant.